alica_engine/include/engine/IAlicaTrace.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <amr_interfaces/TraceContext.h>
+
+namespace alica
+{
+
+class IAlicaTrace
+{
+public:
+    virtual ~IAlicaTrace() = default;
+    IAlicaTrace(const std::string& opName, std::optional<const amr_interfaces::TraceContext> parent = std::nullopt);
```

Comment:
the interface should not have any constructors, because the factory creates the traces. Use the same design as alica timer factory & alica timers